### PR TITLE
Added documentation for operator hub release

### PR DESCRIPTION
**Pull request by rajneesh-ops**
**Description**

This is a PR to add documentation and code changes to publish Eunomia at operator hub.

https://github.com/KohlsTechnology/eunomia/issues/26

**Type of change**

* New feature (non-breaking change which adds functionality)

**Checklist**

- [X] Documentation updated

### CODE REVIEW

File: .github/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+export old_version=<old-version>
+export quay_test_repo=<quay-test-repo>
+export community_fork=<a-fork-of-community-operator>
+```
+
+An example could be:-
+
+```shell
+export new_version=0.1.0
+export old_version=0.1.1
```
Comment (seanmalloy) [inline .github/RELEASE.md]: I think these are backwards. Should the "new version" be 0.1.1, and the "old version" be 0.1.0?

File: .github/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        "username": "'"${QUAY_USERNAME}"'",
+        "password": "'"${QUAY_PASSWORD}"'"
+    }
+}' | jq -r '.token')
+```
+
+Push the catalog to the quay application registry (this is different than a container registry).
+
+```shell
+operator-courier push deploy/olm-catalog/eunomia $quay_test_repo eunomia $new_version "${AUTH_TOKEN}"
```
Comment (seanmalloy) [inline .github/RELEASE.md]: Did we actually do this for v0.1.0? I don't think we have a "quay application registry" setup. Do we?

File: .github/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+Deploy the operator source
+
+```shell
+envsubst < deploy/olm-catalog/operator-source.yaml | oc apply -f -
+```
+
+Now you should see the operator in the operator catalog, follow the normal installation process from here.
```
Comment (seanmalloy) [inline .github/RELEASE.md]: I do not understand what "follow the normal installation process from here" mean? Can you clarify what that means?

File: deploy/olm-catalog/eunomia/0.1.0/eunomia.v0.1.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,286 @@
+apiVersion: operators.coreos.com/v1alpha1
```
Comment (seanmalloy) [inline deploy/olm-catalog/eunomia/0.1.0/eunomia.v0.1.0.clusterserviceversion.yaml]: So this file is that got merged into repo https://github.com/operator-framework/community-operators. Do we actually need to keep this in our repo?

File: deploy/olm-catalog/eunomia/0.1.0/gitopsconfigs.eunomia.kohls.io.crd.yaml [hunk 5]
```diff
@@ -0,0 +1,161 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (seanmalloy) [inline deploy/olm-catalog/eunomia/0.1.0/gitopsconfigs.eunomia.kohls.io.crd.yaml]: Same for this file. This file was submitted to repo https://github.com/operator-framework/community-operators. Why do we need this file in our repo?

File: deploy/olm-catalog/eunomia/eunomia.package.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+packageName: eunomia
```
Comment (seanmalloy) [inline deploy/olm-catalog/eunomia/eunomia.package.yaml]: Same here ... why do we need this file in our repo? The file is in the https://github.com/operator-framework/community-operators repo.

File: deploy/operator.yaml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1
```
Comment (seanmalloy) [inline deploy/operator.yaml]: Why do we need this file in our repo?

File: deploy/role-binding.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# cluster role binding for the operator
```
Comment (seanmalloy) [inline deploy/role-binding.yaml]: Why do we need this file in our repo?

File: deploy/role.yaml [hunk 9]
```diff
@@ -0,0 +1,67 @@
+kind: Role
```
Comment (seanmalloy) [inline deploy/role.yaml]: Why do we need this file in our repo?

File: deploy/service-account.yaml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+# service account for the operator
```
Comment (seanmalloy) [inline deploy/service-account.yaml]: Why do we need this file in our repo?

File: deploy/operator.yaml (hunk 7 above)
Comment (sbar95) [reply deploy/operator.yaml]: As far as I remember the files are needed solely for the `operator-sdk olm-catalog gen-csv --csv-version $new_version --from-version $old_version` command to generate newer csv. If they are not present, the command won't run. I remember I was experimenting with removing some of the files - not all are obligatory (as far as I remember) but I can't recall which ones.

Comment (seanmalloy) [reply deploy/operator.yaml]: But do we need to commit them to our repo? Can't we just generate the files and push them to the https://github.com/operator-framework/community-operators repo?

File: .github/RELEASE.md (hunk 2 above)
Comment (rajneesh-ops) [reply .github/RELEASE.md]: This part was required for testing only so I tested it with mine.

**Comment by seanmalloy** [comment]
@rajneesh-ops can you update `.github/RELEASE.md` with the link to the operator-sdk docs you mentioned this morning?

**Comment by seanmalloy** [comment]
Here is the link to the document, https://github.com/operator-framework/operator-sdk/blob/master/doc/user/olm-catalog/generating-a-csv.md#configuration. Please include this link in file `.github/RELEASE.md`.

**Comment by rajneesh-ops** [comment]
@seanmalloy, I have added the reference link and updated the Release Readme. Let me know if you want me to remove the CSV and other files.

**Comment by seanmalloy** [comment]
@rajneesh-ops please remove the CSV and other files from this pull request. I created #251 to track the automated operatorhub.io deployment.

**Comment by rajneesh-ops** [comment]
@seanmalloy , done with the suggested changes

**Pull request merged**